Enhancing env type built-in variable documentation (#881)
harness/developer-hub: docs/platform/12_Variables-and-Expressions/harness-variables.md

> The environment type, such as production or non-production.

Reviewer: ```suggestion
The environment type, such as `Production` or `PreProduction`.
```

The available values are:

> - PreProduction
> - Production

Reviewer: ```suggestion
- `PreProduction`
- `Production`
```

> You can evaluate it with JEXL expressions in the **Conditional Execution** Settings:

Reviewer: ```suggestion
You can evaluate the expression using JEXL in the **Conditional Execution** settings of steps or stages:
```
